featurebyte/session/base.py

```diff
@@ -87,25 +95,41 @@ def _func_wrapper(func: Any, thread_info: Dict[str, Any], *args: Any, **kwargs:
         thread_info["tid"] = threading.get_ident()
         return func(*args, **kwargs)
 
-    def _raise_timeout_exception_in_thread(thread_id: int) -> None:
+    def _raise_exception_in_thread(thread_id: int) -> None:
         res = ctypes.pythonapi.PyThreadState_SetAsyncExc(
-            ctypes.c_ulong(thread_id), ctypes.py_object(TimeoutError)
+            ctypes.c_ulong(thread_id), ctypes.py_object(asyncio.exceptions.CancelledError)
         )
         if res > 1:
             ctypes.pythonapi.PyThreadState_SetAsyncExc(thread_id, 0)
             raise ValueError("Exception raise failure")
         if res:
-            logger.debug("Raised exception in thread")
+            logger.info("Raised exception in thread")
 
     thread_info: Dict[str, int] = {}
     func_call = functools.partial(ctx.run, _func_wrapper, func, thread_info, *args, **kwargs)
 
     try:
-        return await asyncio.wait_for(loop.run_in_executor(None, func_call), timeout)
-    except asyncio.exceptions.TimeoutError:
+        coro = loop.run_in_executor(None, func_call)
```

Comment:
Check for task completion without using await or wait_for so that task cancellation can be caught